pandas/core/frame.py

```diff
@@ -3223,7 +3223,7 @@ def _setitem_array(self, key, value):
             self._check_setitem_copy()
             self.iloc[indexer] = value
         else:
-            if isinstance(value, DataFrame):
+            if isinstance(value, DataFrame) and self.columns.is_unique:
```

Comment:
will there be a problem on L3230 if value.columns is not unique?

Reply:
No but maybe at 3227

Reply:
Depends of the key references a dup column

Reply:
let me rephrase this:
We don't get there, becasue 3227 would raise. If we would change the condition to recognize duplicates, we would get a problem in 3229, because the lenght of both lists would not match.
If we also fix this, we have to check that in case of a key which is duplicated in the columns, we select two columns and not only one in 3230 on the rhs

Reply:
yah the other case im now unsure of is `not all(x in self.columns for x in key)`

Reply:
Yikes, this has ugly side effects. We have to find a way to handle duplicates there instead of dispatching down :(

And improve the test coverage of these cases